addons/mail/models/mail_followers.py

```diff
@@ -74,6 +74,30 @@ def unlink(self):
     # Private tools methods to fetch followers data
     # --------------------------------------------------
 
+    @api.model
+    def _get_mail_followers(self, mail_ids):
+        """ Get recipients followers of the mails.
+
+        :param list mail_ids: mail_mail ids
+        :return: followers of the related record of the mails limited to the
+            recipients of the mails as a set of tuple (model, res_id, partner_id).
+        :rtype: set
+        """
+        self.env['mail.mail'].flush_model(['message_id', 'recipient_ids'])
+        self.env['mail.followers'].flush_model(['partner_id', 'res_model', 'res_id'])
+        self.env['mail.message'].flush_model(['model', 'res_id'])
+        self.env.cr.execute("""
+                    SELECT message.model, message.res_id, mail_partner.res_partner_id
+                      FROM mail_mail mail        
+                      JOIN mail_mail_res_partner_rel mail_partner ON mail_partner.mail_mail_id = mail.id
+                      JOIN mail_message message ON mail.mail_message_id = message.id
+                      JOIN mail_followers follower ON message.model = follower.res_model 
+                       AND message.res_id = follower.res_id 
+                       AND mail_partner.res_partner_id = follower.partner_id
+                     WHERE mail.id IN %(mail_ids)s 
+                """, {'mail_ids': tuple(mail_ids)})
+        return set(self.env.cr.fetchall())
```

Comment:
Actually, I only use it for the mail, not for the inbox as for inbox I don’t rely on mail_mail but solely on mail_message. Currently, it returns a set of tuple (model, res_id, partner_id) and it is efficient because for each mail, we send, we only check the presence of (message.model, message.res_id, partner_id) where partner_id is the partner for which we personalize the mail. But maybe the returned that is to raw and the form you propose would be easier to use for future development ?

Reply:
That's more a pesonnal taster. Ok to keep the current format :)